c2corg_ui/static/js/editing/articleediting.js

```diff
@@ -53,4 +53,60 @@ app.ArticleEditingController.prototype.filterData = function(data) {
   return data;
 };
 
+/**
+ * @param {appx.Document} doc Document attributes.
+ * @override
+ * @public
+ */
+app.ArticleEditingController.prototype.PreSetQuality = function(doc) {
+  var score = 0;
+  var score_temp = 0;
+  var activities = doc['activities'];
+
+  if ('summary' in doc.locales[0] && doc.locales[0]['summary']) {
+    score = score + 1;
+  }
+ 
+  if (doc['associations']['waypoints'].length > 0 ) {
+    score = score + 1;
+  }
+  
+  if (doc['associations']['images'].length > 0 ) {
+    score = score + 1;
+  }
+  
+  if ('description' in doc.locales[0] && doc.locales[0]['description']) {
+     score = score + 1;
+    
+    var pattern_title = /(^|\n)##/g; // regex for title
+    var result_search_title = doc['locales'][0]['description'].search(pattern_title)
+    if (result_search_title !== -1) {
+      score = score + 1;
+    }
+
+    var pattern_img = /\[img=/g; // regex for image
+    var result_search_img = doc['locales'][0]['description'].search(pattern_img)
+    if (result_search_img !== -1) {
+      score = score + 1;
+    }
+    
+  }
+  else {
+    score = Math.min(score,1);
+  }
+
+  score = Math.min(score,4);
+  
+  if (score === 0){
+    doc['quality'] = 'empty';}
+  else if (score === 1){
+    doc['quality'] = 'draft';}
+  else if (score === 2){
+    doc['quality'] = 'medium';}
+  else if (score === 3){
+    doc['quality'] = 'fine';}
+  else if (score === 4){
+    doc['quality'] = 'great';}
+};
```

Comment:
What about using a ``switch`` instead?
```
switch(score) {
  case 0:
    doc['quality'] = 'empty';
    break;
  // ...
}
```

Reply:
Since this part of code seems to be the same in all controllers, you should probably consider factorizing it somewhere, or have it only in the parent class method.

Reply:
One could also have a map score => quality to avoid the switch